src/west/commands/project.py

```diff
@@ -845,6 +856,25 @@ def do_run(self, args, user_args):
 # Private helper routines.
 #
 
+def _clean_west_refspace(project):
+    # Clean the refs/west space to ensure they do not show up in 'git log'.
+
+    # Get all the ref names that start with refs/west/.
+    list_refs_cmd = ('for-each-ref --format="%(refname)" -- ' +
+                     QUAL_REFS + '**')
+    cp = project.git(list_refs_cmd, capture_stdout=True)
+    west_references = cp.stdout.decode('utf-8').strip()
+
+    # Safely delete each one.
+    for ref in west_references.splitlines():
+        delete_ref_cmd = 'update-ref -d ' + ref
```

Comment:
I'm not sure what "safely" meant here but this very ironically... not safe :-) 4 years later, fix submitted in
- #683